### DISCUSSION TITLE
Home assistant - change unit type

### DISCUSSION THREAD

**Discussion by tim-devel** [john30/ebusd#827] [Q&A] [answered]
Hi.  Is there a way to adjust the device class and unit of measurement for individual entities in mqtt-hassio.cfg.  I am getting the errors below so need to overide the default device class or unit of measurement for some sensors

`2023-02-01 20:19:44.268 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.boiler_modulation (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using native unit of measurement '%' which is not a valid unit for the device class ('temperature') it is using; expected one of ['K', '°F', '°C']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22`
`2023-02-01 20:20:44.608 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.boiler_pump_power (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('power') it is using; expected one of ['W', 'kW']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22`

**Comment by Koky05** [comment, marked as answer]
Check your "mqtt-hassio.cfg" there si part about type and unit definition:

> '# HA integration: the mapping list for (potentially) writable number entities by field type, name, message, and unit.
> type_switch-w-number =
>     number,temperature, = temp|,°C$
>     number,power_factor, = power*%%
>     number,power, = power|,kW$|,W$
>     number,voltage, = volt|,V$
>     number,current, = current,|,A$
>     number,pressure, = bar$
>     number,gas, = gas*/min$
>     number,humidity, = humid*%%$
>     number,, = curve,
> 
> '# HA integration: the mapping list for numeric sensor entities by field type, name, message, and unit.
> type_switch-number =
>     sensor,temperature,measurement = temp|,°C$
>     sensor,power_factor,measurement = power*%%
>     sensor,power,measurement = power|,kW$|,W$
>     sensor,voltage,measurement = volt|,V$
>     sensor,current,measurement = current,|,A$
>     sensor,energy,total_increasing = energy|,Wh$
>     sensor,yield,total_increasing = total*,Wh$
>     sensor,,total_increasing = hours|,h$
>     sensor,,total_increasing = starts*,$
>     sensor,pressure,measurement = bar$
>     sensor,gas,measurement = gas*/min$
>     sensor,humidity,measurement = humid*%%$
>     sensor,, =
> 
> '# HA integration: the mapping list for (potentially) writable binary switch entities by field type, name, message, and unit.
> type_switch-w-list =
>     switch,, = onoff
>     switch,,,yesno = yesno
> 
> '# HA integration: the mapping list for rather binary sensor entities by field type, name, message, and unit.
> type_switch-list =
>     binary_sensor,,measurement = onoff
>     binary_sensor,,measurement,yesno = yesno
>     sensor,, =

**Reply by tim-devel** [reply]
Thanks @Koky05. I have been playing with this for a couple of days and struggling to make it work.  I have posted below the reading I am try to customise and some options I tried on mqtt.cfg but none of them seem to work.  Can you spot what I am doing wrong?  Any advice would be gratefully received!

`r,,ModulationTempDesired,Modulationsetpoint_DK,,,,"2E00",,,SIN,10,%,Modulation set point,,,`

`sensor,power_factor,measurement = modulation*%%`
`sensor,power_factor,measurement = %`
`sensor,power_factor,measurement = *modulation*`

**Reply by tim-devel** [reply]
I think I have solved this.  The code below works but needed to be higher up the list.  ModulationTempDesired was being picked up as a temperature sensor as there was an entry higher up the list that captured the "temp" bit of the name and throught it was a temperature reading.

sensor,power_factor,measurement = modulation*%%
